File: 백준/Silver II/11053. 가장 긴 증가하는 부분 수열/가장 긴 증가하는 부분 수열.py
import sys

N = int(sys.stdin.readline())

Nlist = [0]+list(map(int,sys.stdin.readline().split()))

dp = [0]*(N+1)

for i in range(1,N+1) :
    mx= 0 
    for j in range(0,i) :
        if Nlist[i]>Nlist[j] :
            #새로들어갈 
            mx = max(mx,dp[j])
    dp[i]=mx+1


print(max(dp))

    

File: 백준/Silver/11053. 가장 긴 증가하는 부분 수열/가장 긴 증가하는 부분 수열.java
import java.util.*;
import java.lang.*;
import java.io.*;

// The main method must be in a class named "Main".
class Main {
    
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        int N = Integer.parseInt(br.readLine());

        String [] nstr = br.readLine().split(" ");
        int [] Narr = new int[N];

        for (int i = 0 ; i < N ; i++)
            {
                Narr[i] = Integer.parseInt(nstr[i]);
            }

        int [] dp = new int [N];
        dp[0] = 1;

        for (int i = 1; i< N; i++)
        {
            dp[i] = 1;

            for (int j = 0 ; j< i ; j++)
                {
                    if (Narr[i] > Narr[j])
                    {
                        dp[i] = Math.max(dp[i], dp[j]+1);
                    }
                }
        }

        int answer = 0 ;

        for (int i = 0 ; i < N ; i++)
            {
                answer = Math.max(answer, dp[i]);
            }
        bw.write(String.valueOf(answer));
        bw.close();
    }

}